### ISSUE TITLE
Registered stage writer for mayaUsdProxyShape is not overloadable in python [EMSUSD-1769]

### ISSUE THREAD

**Issue by nanderson-cinesite** [Autodesk/maya-usd#3981]
**Describe the bug**
The recent changes in 0.30.0 (EMUSD-1115 EMUSD-1496) have caused issues with our current pipeline workflow.

Prior to this release there was no official translator for the `mayaUsdProxyShape` node. We wrote our own custom PrimWriter in python to handle various workflows which rely mostly on anonymous layers. Now our custom writer will no longer work as the built-in translator takes priority. 

It also appears that the official translator does not handle anonymous layers as we get the following error:
**Warning: Cannot export the stage in the proxy shape at path '|mayaProxy|mayaProxyShape': the root layer is anonymous**

Our custom writer copied the prim spec from root layer to proxy transform when anonymous layers were used.

We have two other custom PrimWriters written in python that overload default writing (joint and locator) that still work.

We would like to know what you would recommend as a solution for this problem.

**Steps to reproduce**
1. Create simple python based prim writer for mayaUsdProxyShape
2. Register the prim writer in session 
3. Run Export of a stage with anonymous layer
4. Proxy Transform is output with warnings but contents of stage are skipped, i.e. the custom writer is not used.

**Expected behavior**
- The registered custom prim writer should be used over the built-in writer

Maya version: 2024.2
Maya USD version: 0.30.0
USD version: 22.11

**Additional context**

**Comment by santosg87** [comment]
hi @nanderson-cinesite, thank you for reporting this. 

I am not sure i am understanding the workflow you describe. 

When you mention export - are you export maya data (including the stage currently existing on the scene), or am i reading the workflow wrong? 

would you be able to share any of the primwriter samples that are hitting this case for you guys? I am currently investigating this and just trying to better understand the case :)

**Comment by nanderson-cinesite** [comment]
The workflow is likely specific to our pipeline. Basically we use mayaUsdProxyShape nodes with anonymous layers as asset references. We rely on our custom PrimWriter to transfer the USD edits made in the proxy anonymous layer to the final output stage. 

I put together an example of what used to work in 0.29.0 which no longer works in 0.30.0. 

[example.py.txt](https://github.com/user-attachments/files/17549608/example.py.txt)

**Comment by wallworm** [comment]
Thank you for sharing with us. We'll investigate why you cannot override this prim writer.

**Comment by santosd** [comment]
Hi @nanderson-cinesite , it seems this issue will require a fix internally so I have logged an issue on our end so that we can have our devs take a look a finding a solution.

**Comment by samuelliu-adsk** [comment]
Hi @nanderson-cinesite , have you tried using CanExport() to mark the custom primWriter as `Supported`. This way the custom primWriter will have the priority over default ones. A quick example will be this: https://github.com/Autodesk/maya-usd/blob/dev/test/lib/mayaUsd/fileio/testPrimWriter.py#L54

**Comment by nanderson-cinesite** [comment]
I thought I did, I just tested it again and this appears to work.

I wasn't aware that it was required to overload the built in prim writers. In hindsight it makes sense now that there is a built in translator for the proxy shape. 

The other two prim writers I wrote for joints and locators already had `CanExport()` defined. I seem to remember adding this back in February when our unit tests were failing after updating the plugin.

Thanks for your help.
